script/update_beats.sh

```diff
@@ -1,17 +1,24 @@
-#!/usr/bin/env bash -ex
+#!/usr/bin/env bash
+
+set -ex
 
 BEATS_VERSION="${BEATS_VERSION:-master}"
 
 # Find basedir and change to it
-BASEDIR=$(dirname "$0")/../_beats
+DIRNAME=$(dirname "$0")
+BASEDIR=${DIRNAME}/../_beats
 mkdir -p $BASEDIR
-cd $BASEDIR
+pushd $BASEDIR
 
 # Check out beats repo for updating
 GIT_CLONE=repo
-trap "{ rm -rf ${GIT_CLONE}; }" EXIT
+trap "{ rm -rf ${BASEDIR}/${GIT_CLONE}; }" EXIT
```

Comment:
just noticed this doesn't work if exit happens before the `popd`.

Reply:
I've never used `pushd` and `popd` before but from reading what it does, do you even need to use `${BASEDIR}` here again or would it be enough to `rm -rf ${GIT_CLONE}`?

Reply:
I experimented with killing the script in a bunch of places to confirm that the `rm` is executed relative to the working directory pushd leaves the script in before exiting.  One gross fix is to popd and ignore its error within the trap, that might have to do for now.

Reply:
Implemented that in https://github.com/elastic/apm-server/pull/459/commits/5869b6524baeacaaad2858538e7b5b6394e6aa3d